### PR TITLE
CT-1190 Add tomasfejfar/phpstan-phpunit

### PR DESCRIPTION
**Pull request by tomasfejfar**
Jira: CT-1190
KBC: XXX

Before asking for review make sure that:

## Checklist

- [x] New client method(s) has tests
- [x] Apiary file is updated

## Release

  - [x] I gave the PR a proper label:
    * major (BC break)
    * minor (new feature)
    * patch (backwards compatible fix)
    * no release (just test changes)

### CODE REVIEW

**Comment by tomasfejfar** [comment]
Test build https://dev.azure.com/keboola-dev/Connection%20build/_build/results?buildId=41106&view=results

File: composer.json [hunk 1]
```diff
@@ -36,13 +42,14 @@
 		"microsoft/azure-storage-blob": "^1.5",
 		"psr/log": "^1.1|^2.0|^3.0",
 		"symfony/filesystem": "^6.0||^5.0||^4.0",
-		"symfony/process": "^6.0||^5.0||^4.0"
+		"symfony/process": "^6.0||^5.0||^4.0",
+		"tomasfejfar/phpstan-phpunit": "dev-first"
```
Comment (jirkasemmler) [inline composer.json:46]: haha, jsem si nejdirv myslel, ze to je nejakej zase composer specialni pripad, ale ona se jen ta branch jmenuje `first` :D

**Review by jirkasemmler — APPROVED** [review]
bylo by lepsi, kdybys to tam uz mel vydane, ale muzeme zatim zit s timto. dohodli jsme se tak na standupu

**Comment by jirkasemmler** [comment]
btw rebase pls

**Comment by tomasfejfar** [comment]
```diff
git dft
diff --git a/composer.json b/composer.json
index 7e44b226..f51a0607 100644
--- a/composer.json
+++ b/composer.json
@@ -24,12 +24,6 @@
                        "Keboola\\UnitTest\\": "tests-unit/"
                }
        },
-       "repositories": [
-               {
-                       "type": "vcs",
-                       "url": "https://github.com/tomasfejfar/phpstan-phpunit.git"
-               }
-               ],
        "require": {
                "php": ">=7.4",
                "ext-json": "*",
@@ -43,7 +37,7 @@
                "psr/log": "^1.1|^2.0|^3.0",
                "symfony/filesystem": "^6.0||^5.0||^4.0",
                "symfony/process": "^6.0||^5.0||^4.0",
-               "tomasfejfar/phpstan-phpunit": "dev-first"
+               "tomasfejfar/phpstan-phpunit": "~0.0.1"
        },
        "require-dev": {
                "ext-pdo": "*",
diff --git a/phpstan-baseline.neon b/phpstan-baseline.neon
index af73b0e2..d0e93a39 100644
--- a/phpstan-baseline.neon
+++ b/phpstan-baseline.neon
@@ -2235,11 +2235,6 @@ parameters:
                        count: 1
                        path: src/Keboola/StorageApi/TableExporter.php

-               -
-                       message: "#^You should always add `\\$this\\-\\>fail\\(\\)` as a last statement in try/catch block\\.$#"
-                       count: 1
-                       path: tests/Backend/CommonPart1/BranchStorageTest.php
-
                -
                        message: "#^Method Keboola\\\\Test\\\\Backend\\\\CommonPart1\\\\CreateTableTest\\:\\:createTableFromSlicedFileData\\(\\) has no return type specified\\.$#"
                        count: 1
@@ -3250,11 +3245,6 @@ parameters:
                        count: 1
                        path: tests/Backend/MixedSnowflakeSynapse/SharingTest.php

-               -
-                       message: "#^You should always add `\\$this\\-\\>fail\\(\\)` as a last statement in try/catch block\\.$#"
-                       count: 1
-                       path: tests/Backend/MixedSnowflakeSynapse/SharingTest.php
-
                -
                        message: "#^Method Keboola\\\\Test\\\\Backend\\\\Redshift\\\\CreateTableTest\\:\\:syncAsyncData\\(\\) has no return type specified\\.$#"
                        count: 1
@@ -3270,21 +3260,6 @@ parameters:
                        count: 1
                        path: tests/Backend/Redshift/WhereFilterTest.php

-               -
-                       message: "#^You should always add `\\$this\\-\\>fail\\(\\)` as a last statement in try/catch block\\.$#"
-                       count: 1
-                       path: tests/Backend/SOX/BranchStorageTest.php
-
-               -
-                       message: "#^You should always add `\\$this\\-\\>fail\\(\\)` as a last statement in try/catch block\\.$#"
-                       count: 1
-                       path: tests/Backend/SOX/SOXTokensTest.php
-
-               -
-                       message: "#^You should always add `\\$this\\-\\>fail\\(\\)` as a last statement in try/catch block\\.$#"
-                       count: 1
-                       path: tests/Backend/SOX/SOXWorkspaceTest.php
-
                -
                        message: "#^Method Keboola\\\\Test\\\\Backend\\\\Snowflake\\\\CloneIntoWorkspaceTest\\:\\:aliasSettingsProvider\\(\\) has no return type specified\\.$#"
                        count: 1
diff --git a/tests/Backend/CommonPart1/BranchStorageTest.php b/tests/Backend/CommonPart1/BranchStorageTest.php
index 42c11d4f..d6bc668f 100644
--- a/tests/Backend/CommonPart1/BranchStorageTest.php
+++ b/tests/Backend/CommonPart1/BranchStorageTest.php
@@ -30,8 +30,8 @@ public function testCanCreateBucket(): void
         $bucketId = self::STAGE_IN . 'c-' . $this->getTestBucketName($description);
         try {
             $this->_client->getBucket($bucketId);
+            // @phpstan-ignore: tomasfejfar-phpstan-phpunit.missingFailInTryCatch
             $this->_client->dropBucket($bucketId, ['force' => true]);
-            // intentionally don't fail
         } catch (ClientException $e) {
             // ignore if bucket not exists
         }
diff --git a/tests/Backend/MixedSnowflakeSynapse/SharingTest.php b/tests/Backend/MixedSnowflakeSynapse/SharingTest.php
index 7a1437bc..49762014 100644
--- a/tests/Backend/MixedSnowflakeSynapse/SharingTest.php
+++ b/tests/Backend/MixedSnowflakeSynapse/SharingTest.php
@@ -297,8 +297,8 @@ public function testWorkspaceLoadData(
         try {
             $this->_client2->getBucket($unloadBucketId);
             // if bucket exists drop it
+            // @phpstan-ignore: tomasfejfar-phpstan-phpunit.missingFailInTryCatch
             $this->_client2->dropBucket($unloadBucketId, ['force' => true]);
-            // intentionally don't fail
         } catch (ClientException $e) {
             $this->assertEquals("Bucket {$unloadBucketId} not found", $e->getMessage());
         } finally {
diff --git a/tests/Backend/SOX/BranchStorageTest.php b/tests/Backend/SOX/BranchStorageTest.php
index cbfb6c29..b17ca1a0 100644
--- a/tests/Backend/SOX/BranchStorageTest.php
+++ b/tests/Backend/SOX/BranchStorageTest.php
@@ -288,8 +288,8 @@ public function testCrossCrudBucket(array $client1, array $client2): void
         $bucketId = 'in.c-' . $testBucketName;
         // cleanup test bucket
         try {
+            // @phpstan-ignore: tomasfejfar-phpstan-phpunit.missingFailInTryCatch
             $this->getDefaultBranchStorageApiClient()->dropBucket($bucketId, ['async' => true]);
-            // intentionally don't fail
         } catch (ClientException $e) {
             if ($e->getCode() !== 404) {
                 throw $e;
diff --git a/tests/Backend/SOX/SOXTokensTest.php b/tests/Backend/SOX/SOXTokensTest.php
index 5d872bac..64b6f1bf 100644
--- a/tests/Backend/SOX/SOXTokensTest.php
+++ b/tests/Backend/SOX/SOXTokensTest.php
@@ -496,8 +496,8 @@ public function testTokenWithCanCreateJobsFlagCanReadConfigurationsAndCreatePriv
         $elevatedComponentsApi = new Components($this->getDefaultBranchStorageApiClient());

         try {
+            // @phpstan-ignore: tomasfejfar-phpstan-phpunit.missingFailInTryCatch
             $elevatedComponentsApi->deleteConfiguration('wr-db', 'main');
-            // intentionally don't fail
         } catch (ClientException $e) {
             // ignore
         }
diff --git a/tests/Backend/SOX/SOXWorkspaceTest.php b/tests/Backend/SOX/SOXWorkspaceTest.php
index b7e70440..11ebb2f6 100644
--- a/tests/Backend/SOX/SOXWorkspaceTest.php
+++ b/tests/Backend/SOX/SOXWorkspaceTest.php
@@ -257,10 +257,10 @@ public function testTableLoadAsView(): void
             $tableRef = $backend->getTableReflection('languages');
             self::assertEquals(['id', 'name', '_timestamp'], $tableRef->getColumnsNames());
             self::assertCount(10, $backend->fetchAll('languages'));
+            // @phpstan-ignore: tomasfejfar-phpstan-phpunit.missingFailInTryCatch
             if ($backend instanceof BigqueryWorkspaceBackend) {
                 $this->fail('BigQuery should throw exception');
             }
-            // intentionally don't fail
         } catch (ClientException $e) {
             if ($backend instanceof BigqueryWorkspaceBackend) {
                 self::assertSame('Backend "bigquery" does not support: "Other types of loading than view".', $e->getMessage());


``

**Pull request merged**